### ISSUE TITLE
Unusual entity info after conversion from `mmtf` to `pdb` and back

### ISSUE THREAD

**Issue by zacharyrs** [biojava/biojava#931]
Taking an `mmtf`, converting to `pdb`, and then back to `mmtf` yields some funky entity info.

Specifically, take "3zyb", which has 6 entities, the first of is "PA-I galactophilic lectin", with asymIds A through H.
After converting to a `pdb` and back, we have an unnamed first entity (expected), with asymIds as below.
```
A,B,C,D,E,F,G,H,C,D,E,F,G,H,D,E,F,G,H,E,F,G,H,F,G,H,G,H,H
```
I don't think this is desired?

The other entities have the correct number of chains associated (albeit slightly different asymIds, but I think it's a non-issue).

---
<details>
<summary>Code to reproduce</summary>

```java
MmtfUtils.setUpBioJava();
try {
    Structure struc = MmtfActions.readFromFile((new File("/tmp/3zyb.mmtf")).toPath());
    System.out.println(struc.getEntityInfos());
    Files.write((new File("/tmp/3zyb.pdb")).toPath(), struc.toPDB().getBytes());

    Structure struc2 = StructureIO.getStructure((new File("/tmp/3zyb.pdb")).toURI().toURL().toString());
    System.out.println(struc2.getEntityInfos());
    MmtfActions.writeToFile(struc2, (new File("/tmp/3zyb_weird.mmtf")).toPath());
} catch (Exception e) {
    e.printStackTrace();
    System.exit(1);
}
```

</details>

**Comment by josemduarte** [comment]
Thanks for the report. That looks like a bug in `EntityFinder.findPolyEntities()`, which is used when reading from PDB file when no COMPND records are present. Entities are found by sequence alignment of chains. I'll see if I can have a look.

In any case, I would recommend in general to use mmCIF as the way to write out. The PDB format has been legacy for a long time.

**Comment by josemduarte** [comment]
See fix in #938 . This was purely a PDB-format reading issue. `EntityFinder.findPolyEntities()` wasn't working properly for entities with more than 3 chains.
